Sources/Bond/Collections/ObservableArray.swift

```diff
@@ -295,30 +296,30 @@ extension MutableObservableArray {
 }
 
 extension MutableObservableArray where Item: Equatable {
-  
+
   public func replace(with array: [Item], performDiff: Bool) {
-    if performDiff {
-      lock.lock()
+    lock.lock(); defer { lock.unlock() }
 
+    if performDiff {
       let diff = self.array.extendedDiff(array)
+      let patch = diff.patch(from: self.array, to: array)
```

Comment:
There are two ways as far as I can see...

1. Make use of type system so it's impossible to bind signal (array) with wrong batching algorithm. That would involve new types, probably another kind of `ObservableArrayEvent`. Our operator would then return `Signal<AlternativeObservableArrayEvent<Item>, Error>`. Alternative type would be identical to the original one, but NS views would only accept bindings with it. If we do that, compiler would guaranty that NS bindings work because `AlternativeObservableArrayEvent` events are generated only by `patchedBatch` operator. 
2. Only document it.

Problem with (1) is that it makes things more complex, while (2) makes it easy to do the wrong thing. My preference is (1). Name `AlternativeObservableArrayEvent` and `patchedBatch` are not good though 😄